[clang-tidy] Add type annotations to add_new_check.py, fix minor bug

nicovank opened the pull request:
```
> python3 -m mypy --strict clang-tools-extra/clang-tidy/add_new_check.py
Success: no issues found in 1 source file
```

Also fix a bug when `--standard` is not provided on the command line: the
generated test case has a `None` causing issues:
```
> python3 clang-tools-extra/clang-tidy/add_new_check.py performance XXX
Updating clang-tools-extra/clang-tidy/performance/CMakeLists.txt...
Creating clang-tools-extra/clang-tidy/performance/XxxCheck.h...
Creating clang-tools-extra/clang-tidy/performance/XxxCheck.cpp...
Updating clang-tools-extra/clang-tidy/performance/PerformanceTidyModule.cpp...
Updating clang-tools-extra/docs/ReleaseNotes.rst...
Creating clang-tools-extra/test/clang-tidy/checkers/performance/XXX.cpp...
Creating clang-tools-extra/docs/clang-tidy/checks/performance/XXX.rst...
Updating clang-tools-extra/docs/clang-tidy/checks/list.rst...
Done. Now it's your turn!

> head -n 1 clang-tools-extra/test/clang-tidy/checkers/performance/XXX.cpp
// RUN: %check_clang_tidy None%s performance-XXX %t
```

nicovank commented on clang-tools-extra/clang-tidy/add_new_check.py:
```diff
@@ -324,9 +329,14 @@ def add_release_notes(module_path, module, check_name, description):
+    test_extension: str,
+    test_standard: Optional[str],
+) -> None:
+    test_standard = f"-std={test_standard}-or-later " if test_standard else ""
```
This is the minor bug fix. If `test_standard` is `None`, it is wrongly formatted into the string a few lines down.

5chmidti commented on clang-tools-extra/clang-tidy/add_new_check.py:
```diff
@@ -754,7 +765,7 @@ def main():
             f"%(lang)s.{cpp_language_to_requirements.get(args.standard, 'CPlusPlus')}"
         )
-    elif language in ["objc", "objc++"]:
+    else:  # "objc" or "objc++"
         language_restrict = "%(lang)s.ObjC"
```
This change would result in any language string that is not supported setting the language restriction to `ObjC` (e.g., `--language CUDA`).
I'd say keep the `elif` and do a
```c++
if language_restrict is None:
    print(f"Unsupported language '{language}' was specified")
```
afterward

nicovank replied:
Other languages technically can not get to this point because of `choices = language_to_extension.keys()` which only contains `["c", "c++", "objc", "objc++"]`, but I agree keeping the condition is more robust in case this lists change.

nicovank commented:
Thanks for review! Addressed comments.

I have no commit access, if/when this looks good to go you can go ahead and click merge.

nicovank committed:
[clang-tidy] Add type annotations to add_new_check.py, fix minor bug

```
> python3 -m mypy --strict clang-tools-extra/clang-tidy/add_new_check.py
Success: no issues found in 1 source file
```

Also fix a bug when `--standard` is not provided on the command line: the
generated test case has a `None` causing issues:
```
> python3 clang-tools-extra/clang-tidy/add_new_check.py performance XXX
Updating clang-tools-extra/clang-tidy/performance/CMakeLists.txt...
Creating clang-tools-extra/clang-tidy/performance/XxxCheck.h...
Creating clang-tools-extra/clang-tidy/performance/XxxCheck.cpp...
Updating clang-tools-extra/clang-tidy/performance/PerformanceTidyModule.cpp...
Updating clang-tools-extra/docs/ReleaseNotes.rst...
Creating clang-tools-extra/test/clang-tidy/checkers/performance/XXX.cpp...
Creating clang-tools-extra/docs/clang-tidy/checks/performance/XXX.rst...
Updating clang-tools-extra/docs/clang-tidy/checks/list.rst...
Done. Now it's your turn!

> head -n 1 clang-tools-extra/test/clang-tidy/checkers/performance/XXX.cpp
// RUN: %check_clang_tidy None%s performance-XXX %t
```
